posts/2024-05-03-simplifying-nosql-database-integration-with-jakarta-nosql.adoc

```diff
@@ -0,0 +1,417 @@
+
+[source, java]
+----
+import jakarta.nosql.Entity;
+import jakarta.nosql.Id;
+import jakarta.nosql.Column;
+
+@Entity
+public class Book {
+
+    @Id
+    private String isbn;
+
+    @Column
+    private String title;
+
+    @Column
+    private String author;
+
+    @Convert(YearConverter.class)
+    @Column
+    private Year year;
+
+}
+----
+
+When using Java 17+, Eclipse JNoSQL allows the usage of Java Records as entities:
+
+[source, java]
+----
+import jakarta.nosql.Entity;
+import jakarta.nosql.Id;
+import jakarta.nosql.Column;
+
+@Entity
+public record Book(@Id String isbn,
+                   @Column("title") String title,
+                   @Column("author") String author,
+                   @Convert(YearConverter.class) @Column("year") Year year,
+                   @Column("edition") int edition) {
+
+}
+
+----
+
+Last but not least, Eclipse JNoSQL, as a Jakarta Data implementation, allows us to create repositories, enabling Domain-Driven Development (DDD) capabilities through the usage of the Repository pattern. This approach simplifies developers who seek to bring the code closer to the business (domain-centric) instead of working with database semantics.
+
+[source,java]
+----
+import jakarta.data.page.Page;
+import jakarta.data.page.PageRequest;
+import jakarta.data.repository.Delete;
+import jakarta.data.repository.Repository;
+import jakarta.data.repository.DataRepository;
+import jakarta.data.repository.Query;
+import jakarta.data.repository.Param;
+import jakarta.data.repository.Save;
+
+@Repository
+public interface Garage extends DataRepository<Car,String>{
+
+    @Save
+    Car park(Car car);
+
+    @Delete
+    Car unPark(Car car);
+
+    @Query("select * from Car where driver.name = @name")
+    Set<Car> findByDriver(@Param("name") String name);
+
+    Page<Car> findByColor(Color color, PageRequest pageRequest);
+
+}
+----
+
+==== What to expect from Eclipse JNoSQL
+
+Beyond being a Jakarta NoSQL and Jakarta Data implementation, such framework intents to reaching out these goals:
+
+* Increase productivity performing common NoSQL operations
+* Use of Convention Over Configuration
+* Rich Object Mapping integrated with Contexts and Dependency Injection (CDI)
+* Java-based Query and Fluent-API
+* Persistence lifecycle events
+* Low-level mapping using Standard NoSQL APIs
+* Specific template API to each NoSQL category
+* Annotation-oriented using JPA-like naming when it makes sense
+* Extensible to explore the particular behavior of a NoSQL database
+* Explore the popularity of Apache TinkerPop in Graph API
+
+== Next Steps: Continuing the Journey
+
+Congratulations on getting this far!
+
+After getting an overview about the Jakarta NoSQL, Jakarta Data, and the Eclipse JNoSQL, I invite you to explore these tools on a hands-on approach, managing and querying data from NoSQL databases and switching between NoSQL databases as needed.
+
+// This blog post is the 1st part of a set of blog posts:
```

Comment:
we'll sort these links as we release the posts but for now maybe something like:

> This blog post is the 1st part of a set of blog posts, keep you eyes out for these follow-ups coming soon:

> * Jakarta NoSQL in Action: JNopo Game

> * Jakarta NoSQL in Action: Switching NoSQL Databases with Ease

Then whenever we publish each post, I'll go back and add the appropriate link (and remove "coming soon"

Reply:
@dmuelle thanks for letting me know about it! I'll change it ASAP and let you know about it.

Reply:
@dmuelle I've changed according to your recommendation!